src/main/kotlin/org/jitsi/jicofo/conference/source/ValidatingConferenceSourceMap.kt

```diff
@@ -177,16 +179,20 @@ class ValidatingConferenceSourceMap(
     }
 
     /** Override [add] to keep the additional [ssrcToOwnerMap] and [msidToOwnerMap] maps updated. */
-    override fun add(other: ConferenceSourceMap) = super.add(other).also {
-        other.forEach { (owner, endpointSourceSet) -> sourceSetAdded(owner, endpointSourceSet) }
+    override fun add(other: ConferenceSourceMap) = synchronized(syncRoot) {
+        super.add(other).also {
+            other.forEach { (owner, endpointSourceSet) -> sourceSetAdded(owner, endpointSourceSet) }
+        }
     }
 
     /** Override [add] to keep the additional [ssrcToOwnerMap] and [msidToOwnerMap] maps updated. */
-    override fun add(owner: Jid?, endpointSourceSet: EndpointSourceSet) = super.add(owner, endpointSourceSet).also {
-        sourceSetAdded(owner, endpointSourceSet)
+    override fun add(owner: Jid?, endpointSourceSet: EndpointSourceSet) = synchronized(syncRoot) {
+        super.add(owner, endpointSourceSet).also {
+            sourceSetAdded(owner, endpointSourceSet)
+        }
     }
 
-    private fun sourceSetAdded(owner: Jid?, endpointSourceSet: EndpointSourceSet) {
+    private fun sourceSetAdded(owner: Jid?, endpointSourceSet: EndpointSourceSet) = synchronized(syncRoot) {
```

Comment:
Is `sourceSetAdded` ever called in a non-synchronized context?  If not you shouldn't need to synchronize it again.  (And if it is, that seems like it could be a race.). Likewise for `sourceSetRemoved`.

Reply:
My preference is to keep the redundant lock aquisition to make it clear that the function requires the lock. I think the performance impact is negligible (we had a similar discussion a long time ago and I ran some basic tests, but I don't have the results anymore). But I'm open to changing it if you insist.

Reply:
Ok.